src/TurbulenceClosures/turbulence_closure_implementations/Smagorinskys/dynamic_coefficient.jl

```diff
@@ -42,6 +42,54 @@ in the `x` and `y` directions).
 
 `DynamicCoefficient` is updated according to `schedule`, and `minimum_numerator` defines the minimum
 value that is acceptable in the denominator of the final calculation.
+
+Examples
+========
+
+```jldoctest
+julia> using Oceananigans
+
+julia> dynamic_coeff = DynamicCoefficient(averaging=(1, 2))
+DynamicCoefficient with
+├── averaging = (1, 2)
+├── schedule = IterationInterval(1, 0)
+└── minimum_numerator = 1.0e-32
+
+julia> dynamic_smagorinsky = Smagorinsky(coefficient=dynamic_coeff)
+Smagorinsky closure with
+├── coefficient = DynamicCoefficient(averaging = (1, 2), schedule = IterationInterval(1, 0))
+└── Pr = 1.0
+```
+
+The dynamic Smagorinsky above has its dynamic coefficient recalculated at every time
+step. While this provides the highest level of accuracy, it will almost certainly be very slow
+given the high cost of calculating `DynamicCoefficient`s. Because of this slowdown, it is common in the
+literature to recalculate the coefficient only every few time steps, with the assumption that the
+their values don't change much from one time-step to the other. Usually the update frequency chosen
```

Comment:
If we say "usually" then we need citation. Eg wikipedia might say [citation needed]. I think a better practice is to give a range so we don't multiply the "collective intellectual debt" that always choosing the same number is having on this community.

Reply:
I'd say try strides of 2-8 or something.

Reply:
I can definitely add some citations. The thing is that I have never seen any other update frequency used in any paper. If you have info on papers that do it differently, let me know and we can give a range. Otherwise, the only number that has been used (afaik) is 5.

And again, I understand the rationale behind going by factors of 2, but it misses the one number that has been widely tested haha.

So please send some papers with different update frequencies my way, but otherwise I'd address your comment by keeping the text the same and adding references so that users can follow up and take their own conclusions.

Reply:
Added citations and made an accurate assessment if the current status: namely that the only values used so far in the literature is every 5 time steps but other values are possible. This way the user can decide what to do by themselves.